src/js/view/preview.js

```diff
@@ -30,6 +35,25 @@ Object.assign(Preview.prototype, {
             backgroundImage: backgroundImage
         });
     },
+    enableZoom: function() {
+        if (this.model.get('state' !== 'idle')) {
+            return;
+        }
+        
+        setTimeout(() => {
+            const originX = Math.ceil(Math.random() * 100) + '%';
+            const originY = Math.ceil(Math.random() * 100) + '%';
+            
+            this.el.classList.add('jw-ab-zoom-thumbnail');
+            this.el.style.transformOrigin = originX + ' ' + originY;
+        }, 2000);
+        
+        this.model.once('change:state', this.destroy, this);
```

Comment:
why do we destroy the preview on state change? i understand we're moving from idle to playing but unclear on why this wasn't needed before

Reply:
I did it to ensure the animation stopped since the element remains and because there was no longer a need for the event listeners. However, I can also just remove the zoomThumbnail class.

Reply:
hmmm yeah i might opt for that - when a use selects `jwplayer().stop()` it's going to put the player back into the idle state so we may want to keep it around for that?

thoughts @robwalch

Reply:
I agree. Because they will also get removed when the preview is destroyed

Reply:
Adding a destroy method is good, but I would prefer that the `once('change:state'` callback run something like `this.stopZoom` or `this. pauseZoom` and you just call that method in `destroy`.

And, since the state can go from IDLE to something else, and then back to IDLE, this could be an `on` handler that checks state. ⚠️  However! it is not efficient to handle all state changes in UI elements. The view is already doing that and debouncing things that would change on buffering, so call these methods from the view for optimal performance.  See  `function _stateUpdate(state) {` in view.js